wren-ui/src/apollo/server/services/dashboardService.ts

```diff
@@ -63,6 +76,57 @@ export class DashboardService implements IDashboardService {
     this.dashboardRepository = dashboardRepository;
   }
 
+  public async setDashboardSchedule(
+    dashboardId: number,
+    data: SetDashboardCacheData,
+  ): Promise<Dashboard> {
+    try {
+      const { cacheEnabled, schedule } = data;
+      // Validate input
+      this.validateScheduleInput(data);
+
+      // Check if dashboard exists
+      const dashboard = await this.dashboardRepository.findOneBy({
+        id: dashboardId,
+      });
+      if (!dashboard) {
+        throw new Error(`Dashboard with id ${dashboardId} not found`);
+      }
+      if (!cacheEnabled) {
+        return await this.dashboardRepository.updateOne(dashboardId, {
+          cacheEnabled: false,
+          scheduleFrequency: null,
+          scheduleTimezone: null,
+          scheduleCron: null,
+          nextScheduledAt: null,
+        });
+      }
+      // Initialize schedule-related variables
+      let cronExpression: string | null = null;
+      let nextScheduledAt: Date | null = null;
+
+      // Process schedule if caching is enabled
+      if (cacheEnabled && schedule.frequency !== ScheduleFrequencyEnum.NEVER) {
+        cronExpression = this.generateCronExpression(schedule);
+        if (cronExpression) {
+          nextScheduledAt = this.calculateNextRunTime(cronExpression);
+        }
```

Comment:
I'm curious under what kind of situaion will `cronExpression` be null or undefined ?